CONTRIBUTING.md

```diff
@@ -67,7 +67,7 @@ We use and recommend the following workflow:
     - The next official build will automatically include your change.
     - You can delete the branch you used for making the change.
 
-Note: It is OK for your PR to include a large number of commits. Once your change is accepted, you will be asked to squash your commits into one or some appropriately small number of commits before your PR is merged.
+Note: It is OK for your PR to include a large number of commits. Once your change is accepted, you will be asked to squash your commits into one or some appropriately small number of commits before your PR is merged. Often it is not necessary to merge multiple commits, then it's preferable to not manually squash commits before merge, as typically the "Squash and merge" option in GitHub would be used to perform both in one step.
```

Comment:
> you will be asked to squash your commits into one or some appropriately small number of commits before your PR is merged

This is obsolete guidance. It was needed before github had "squash & merge" option. We should just delete it.

(Also, this is somewhat redundant with https://github.com/dotnet/runtime/blob/master/docs/pr-guide.md#quick-code-review-rules but that's ok.)